test/diamond/DiamondCutFacet.t.sol

```diff
@@ -0,0 +1,546 @@
+        functionSelectors[0] = bytes4(keccak256("decimals()"));
+
+        DiamondCutFacet.FacetCut[] memory _cutToAddImmutableFunction = new DiamondCutFacet.FacetCut[](1);
+        _cutToAddImmutableFunction[0] = DiamondCutFacet.FacetCut({
+            facetAddress: address(facet),
+            action: DiamondCutFacet.FacetCutAction.Add,
+            functionSelectors: functionSelectors
+        });
+
+        address _init = ADDRESS_ZERO;
+        bytes memory _calldata = bytes("0x0");
+
+        vm.prank(owner);
+        facet.diamondCut(_cutToAddImmutableFunction, _init, _calldata);
+
+        DiamondCutFacet.FacetCut[] memory _cutToReplaceImmutableFunction = new DiamondCutFacet.FacetCut[](1);
+        _cutToReplaceImmutableFunction[0] = DiamondCutFacet.FacetCut({
+            facetAddress: address(token),
+            action: DiamondCutFacet.FacetCutAction.Replace,
+            functionSelectors: functionSelectors
+        });
+
+        vm.prank(owner);
+        vm.expectRevert(
+            abi.encodeWithSelector(DiamondCutFacet.CannotReplaceImmutableFunction.selector, functionSelectors[0])
+        );
+        facet.diamondCut(_cutToReplaceImmutableFunction, _init, _calldata);
+    }
+
+    function test_DiamondCut_initalizeFunctionExecutionWithoutCode() public {
+        (DiamondCutFacet.FacetCut[] memory _cut,, bytes memory _calldata) = _basicAction();
+
+        address _init = address(1);
+
+        vm.prank(owner);
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                DiamondCutFacet.NoBytecodeAtAddress.selector, _init, "DiamondCutFacet: _init address no code"
+            )
+        );
+        facet.diamondCut(_cut, _init, _calldata);
+    }
+
+    function test_DiamondCut_initializeCallWithWrongCalldata() public {
+        (DiamondCutFacet.FacetCut[] memory _cut,,) = _basicAction();
+
+        ERC20FacetHarness newFacet = new ERC20FacetHarness();
+        address _init = address(newFacet);
+
+        bytes memory _wrongCalldata = abi.encodeWithSelector(bytes4(keccak256("doesNotExist(uint256)")), uint256(123));
+
+        vm.prank(owner);
+        vm.expectRevert(
+            abi.encodeWithSelector(DiamondCutFacet.InitializationFunctionReverted.selector, _init, _wrongCalldata)
+        );
+        facet.diamondCut(_cut, _init, _wrongCalldata);
+    }
+
+    function test_DiamondCut_initalizeCallWithWrongCalldataReturningErrorMessage() public {
+        (DiamondCutFacet.FacetCut[] memory _cut,,) = _basicAction();
+
+        ERC20FacetHarnessWithFallback newFacet = new ERC20FacetHarnessWithFallback();
+        address _init = address(newFacet);
+
+        bytes memory _wrongCalldata = abi.encodeWithSelector(
+            bytes4(keccak256("initialize(string,string,uint256)")), TOKEN_NAME, TOKEN_DECIMALS, TOKEN_DECIMALS
+        );
+
+        vm.prank(owner);
+        vm.expectRevert(abi.encode("WRONG FUNCTION CALL"));
+        facet.diamondCut(_cut, _init, _wrongCalldata);
+    }
```

Comment:
I'm probably missing something, but can you explain what this test is testing for?

The revert comes from the ERC20FacetHarnessWithFallbackHarness (not an actual src contract), so it seems to me like this test doesn't actually hit a real contract.